app/src/main/java/tech/relaycorp/ping/domain/SendPing.kt

```diff
@@ -0,0 +1,76 @@
+package tech.relaycorp.ping.domain
+
+import kotlinx.coroutines.flow.first
+import tech.relaycorp.awaladroid.AwaladroidException
+import tech.relaycorp.awaladroid.messaging.ParcelId
+import tech.relaycorp.ping.awala.*
+import tech.relaycorp.ping.data.database.dao.PingDao
+import tech.relaycorp.ping.data.database.entity.PingEntity
+import tech.relaycorp.ping.data.preference.AppPreferences
+import tech.relaycorp.ping.domain.model.Peer
+import java.time.ZonedDateTime
+import javax.inject.Inject
+import kotlin.time.Duration
+
+class SendPing
+@Inject constructor(
+    private val appPreferences: AppPreferences,
+    private val firstPartyEndpointLoad: FirstPartyEndpointLoad,
+    private val publicThirdPartyEndpointLoad: PublicThirdPartyEndpointLoad,
+    private val sendGatewayMessage: SendGatewayMessage,
+    private val pingSerialization: PingSerialization,
+    private val outgoingMessageBuilder: OutgoingMessageBuilder,
+    private val pingDao: PingDao
+) {
+
+    @Throws(SendPingException::class)
+    suspend fun send(peer: Peer, duration: Duration): String {
+        val senderAddress = appPreferences.firstPartyEndpointAddress().first()
+            ?: throw SendPingException("Sender not set")
+        val sender = firstPartyEndpointLoad.load(senderAddress)
+            ?: throw SendPingException("Sender not registered")
+
+        val recipient = publicThirdPartyEndpointLoad.load(peer.privateAddress)
+            ?: throw SendPingException("Recipient not imported")
+
+        val parcelId = ParcelId.generate()
```

Comment:
I think we shouldn't be using the `ParcelId` here: The parcel id and the ping id are completely independent of each other -- they just happen to be UUID4 values.

I'd suggest to let awaladroid generate a random parcel id for the message, and set the ping id here by calling something like `UUID.randomUUID().toString()`.

Thoughts?